crates/editor/src/signature_help_popover.rs

```diff
@@ -0,0 +1,553 @@
+                        documentation: None,
+                    },
+                ]),
+                active_parameter: None,
+            }],
+            active_signature: Some(0),
+            active_parameter: Some(1),
+        };
+        let maybe_markdown = create_signature_help_markdown_string(signature_help);
+        assert!(maybe_markdown.is_some());
+
+        let markdown = maybe_markdown.unwrap();
+        assert_eq!(
+            markdown,
+            (
+                "foo: u8, bar: &str".to_string(),
+                vec![(9..18, SIGNATURE_HELP_HIGHLIGHT)]
+            )
+        );
+    }
+
+    #[test]
+    fn test_create_signature_help_markdown_string_3() {
+        let signature_help = SignatureHelp {
+            signatures: vec![
+                SignatureInformation {
+                    label: "fn test1(foo: u8, bar: &str)".to_string(),
+                    documentation: None,
+                    parameters: Some(vec![
+                        lsp::ParameterInformation {
+                            label: lsp::ParameterLabel::Simple("foo: u8".to_string()),
+                            documentation: None,
+                        },
+                        lsp::ParameterInformation {
+                            label: lsp::ParameterLabel::Simple("bar: &str".to_string()),
+                            documentation: None,
+                        },
+                    ]),
+                    active_parameter: None,
+                },
+                SignatureInformation {
+                    label: "fn test2(hoge: String, fuga: bool)".to_string(),
+                    documentation: None,
+                    parameters: Some(vec![
+                        lsp::ParameterInformation {
+                            label: lsp::ParameterLabel::Simple("hoge: String".to_string()),
+                            documentation: None,
+                        },
+                        lsp::ParameterInformation {
+                            label: lsp::ParameterLabel::Simple("fuga: bool".to_string()),
+                            documentation: None,
+                        },
+                    ]),
+                    active_parameter: None,
+                },
+            ],
+            active_signature: Some(0),
+            active_parameter: Some(0),
+        };
+        let maybe_markdown = create_signature_help_markdown_string(signature_help);
+        assert!(maybe_markdown.is_some());
+
+        let markdown = maybe_markdown.unwrap();
+        assert_eq!(
+            markdown,
+            (
+                "foo: u8, bar: &str (+1 overload)".to_string(),
+                vec![
+                    (0..7, SIGNATURE_HELP_HIGHLIGHT),
+                    (19..32, SIGNATURE_HELP_OVERLOAD_HIGHLIGHT)
+                ]
+            )
+        );
+    }
+
+    #[test]
+    fn test_create_signature_help_markdown_string_4() {
```

Comment:
There's a lot of rendering tests, which is amazing, thank you!

What's lacking are new ones in the `editor_tests.rs`, and after we deal with the brackets and whatever else that triggers the pop-up display, it would be fantastic to have that logic tested too.